ui/marketplace/MarketplaceAppModal.tsx

```diff
@@ -61,8 +73,27 @@ const MarketplaceAppModal = ({
   }
 
   const handleFavoriteClick = useCallback(() => {
-    onFavoriteClick(data.id, isFavorite);
-  }, [ onFavoriteClick, data.id, isFavorite ]);
+    onFavoriteClick(id, isFavorite, 'App modal');
+  }, [ onFavoriteClick, id, isFavorite ]);
+
+  const showContractList = useCallback((type: ContractListTypes) => {
+    onClose();
+    showContractListDefault(id, type);
```

Comment:
For me it will be more clear if it has name `showContractListProp` or `showContractListBase`.